app/src/main/java/com/gatheringhallstudios/mhworlddatabase/common/BaseHubFragment.java

```diff
@@ -56,25 +67,58 @@ public void onActivityCreated(@Nullable Bundle savedInstanceState) {
         getActivity().setTitle(getHubTitle());
     }
 
-    public class GenericPagerAdapter extends FragmentPagerAdapter {
+    /**
+     * Defines an interface for a class that builds a fragment for a tab.
+     */
+    public interface TabBuilder {
+        Fragment build();
+    }
+
+    /**
+     * Internal class to encapsulate a tab's metadata
+     */
+    private class HubTab {
+        public String title;
+        public TabBuilder builder;
+        public HubTab(String name, TabBuilder builder) {
+            this.title = name;
+            this.builder = builder;
+        }
+        public Fragment build() {
+            return builder.build();
+        }
+    }
+
+    /**
+     * Internal class that handles the actual logic of rendering tabs
+     */
+    private class GenericPagerAdapter extends FragmentPagerAdapter {
 
         GenericPagerAdapter(FragmentManager fm) {
             super(fm);
         }
 
         @Override
         public Fragment getItem(int index) {
-            return getTab(index);
+            try {
+                return tabs.get(index).build();
+            } catch (ArrayIndexOutOfBoundsException ex) {
+                return null;
```

Comment:
There should be a log here. You might want to just have a handleError method that prints a log.

Reply:
Took the behavior of the original version, but yeah, that was bothering me. I feel like it should just throw the exception, but a log would be a decent second choice.